### ISSUE TITLE
dubbo 2.7.6 接口分组异常

### ISSUE THREAD

**Issue by tonyLee1** [apache/dubbo#6603]
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [ ] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.7.6
* Java version: 1.8

### Steps to reproduce this issue

1. @Reference(version = "xxx", group = "xxx") , group：xxx 有效
2. @Reference(version = "xxx", group = "xxx1") , group：xxx1 不生效
3. 异常信息：No provider available from registry

**Comment by liangyuchun** [comment]
@Service(group = "xxx1")

**Comment by tonyLee1** [comment]
> @service（组=“ xxx1”）

provider对应有：

@service（版本=“ xxx”，组=“ xxx”）
@service（版本=“ xxx”，组=“ xxx1”）

两个实现类 指向 同一个接口， 通过分组区分

**Comment by TimYi** [comment]
我也遇到了group相关的问题，我遇到的问题是所有group都调用到同一个值了。

**Comment by xiaoheng1** [comment]
can you provider a demo? My test is good.

service：

```
<bean id="groupAService" class="org.apache.dubbo.samples.group.impl.GroupAServiceImpl"/>

    <bean id="groupBService" class="org.apache.dubbo.samples.group.impl.GroupBServiceImpl"/>

    <dubbo:service group="groupA" interface="org.apache.dubbo.samples.group.api.GroupService" ref="groupAService"/>

    <dubbo:service group="groupB" interface="org.apache.dubbo.samples.group.api.GroupService" ref="groupBService"/>

```
client：

```
<dubbo:reference group="groupA" id="groupAService" check="false"
                     interface="org.apache.dubbo.samples.group.api.GroupService"/>

    <dubbo:reference group="groupB" id="groupBService" check="false"
                     interface="org.apache.dubbo.samples.group.api.GroupService"/>
```

**Comment by xiaoheng1** [comment]
Let me have a look

**Comment by xiaoheng1** [comment]
This problem has been fixed in version 2.7.8, please upgrade dubbo version.

**Comment by tonyLee1** [comment]
收到，我升级试试，谢谢你！



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: xiaoheng1 <notifications@github.com&gt;
发送时间: 2021年3月6日 21:38
收件人: apache/dubbo <dubbo@noreply.github.com&gt;
抄送: tonyLee1 <498959412@qq.com&gt;, Author <author@noreply.github.com&gt;
主题: 回复：[apache/dubbo] dubbo 2.7.6 接口分组异常 (#6603)





 
This problem has been fixed in version 2.7.8, please upgrade dubbo version.

**Comment by xiaoheng1** [comment]
&READY-TO-CLOSE&
